app/scripts/views/connect_another_device.js

```diff
@@ -0,0 +1,182 @@
+
+      return {
+        canSignIn: this._canSignIn(),
+        email,
+        escapedSignInUrl,
+        escapedSupportUrl
+      };
+    },
+
+    /**
+     * Get the user-agent string. For functional testing
+     * purposes, first attempts to fetch a UA string from the
+     * `forceUA` query parameter, if that is not found, use
+     * the browser's.
+     *
+     * @returns {String}
+     * @private
+     */
+    _getUserAgentString () {
+      return this.getSearchParam('forceUA') || this.window.navigator.userAgent;
+    },
+
+    /**
+     * Get a user-agent parser instance.
+     *
+     * @returns {Object}
+     * @private
+     */
+    _getUap () {
+      if (! this._uap) {
+        const userAgent = this._getUserAgentString();
+        this._uap = new UserAgent(userAgent);
+      }
+      return this._uap;
+    },
+
+    /**
+     * Check if the current user can sign in.
+     *
+     * @returns {Boolean}
+     * @private
+     */
+    _canSignIn () {
+      // Only users that are not signed in can do so.
+      return ! this.model.get('isSignedIn') &&
+               this._hasWebChannelSupport();
+    },
+
+    /**
+     * Check if the current browser has web channel support.
+     *
+     * @returns {Boolean}
+     * @private
+     */
+    _hasWebChannelSupport () {
+      const uap = this._getUap();
+
+        // All Foxes >= 40 except iOS can sign in.
+      return uap.isFirefox() &&
+           ! uap.isIos() &&
+             uap.browser.version >= 40;
+    },
+
+    /**
+     * Return the sign in context that can be used to sign in
+     * to Sync. Assumes the context is only used if the user
+     * can actually sign in to Sync.
+     *
+     * @returns {String}
+     * @private
+     */
+    _getSignInContext() {
+      if (this._getUap().isAndroid()) {
+        return Constants.FX_FENNEC_V1_CONTEXT;
+      } else {
+        // desktop_v3 is safe for all desktop versions that can
+        // use WebChannels. The only difference between v2 and v3
+        // was the Sync Preferences button, which has since
+        // been disabled.
+        return Constants.FX_DESKTOP_V3_CONTEXT;
+      }
+    },
+
+    /**
+     * Get the SUMO url for "Why is this happening to me"?
+     *
+     * @returns {String}
+     * @private
+     */
+    _getSupportUrl () {
+      // TODO - need to get a support article written!
+      return 'https://support.mozilla.org';
+    },
+
+    /**
+     * Get an escaped sign in URL.
+     *
+     * @param {String} context - context to use to sign in
+     * @param {String} email - users email address, used to
+     *  pre-fill the signin page.
+     * @returns {String}
+     * @private
+     */
+    _getEscapedSignInUrl (context, email) {
+      const origin = this.window.location.origin;
+
+      const params = {
+        context,
+        email,
+        service: Constants.SYNC_SERVICE,
+        utm_source: View.UTM_SOURCE //eslint-disable-line camelcase
```

Comment:
Maybe a better option is to pass all the current utm_* query parameters verbatim, and use `entrypoint` instead of `utm_source`